src/services/analytics/timescaledb/timescaledb.query.service.ts

```diff
@@ -108,32 +119,71 @@ export class TimescaleDBQueryService implements AnalyticsQueryInterface {
                 .addSelect('locf(last(last, time)) as last')
                 .where('series = :series', { series })
                 .andWhere('key = :metric', { metric })
-                .andWhere('time between :start and now()', {
-                    start: startDate,
+                .andWhere('time between :startDate and :endDate', {
+                    startDate,
+                    endDate,
                 })
                 .groupBy('day')
                 .getRawMany();
-            const results =
-                query?.map(
-                    (row) =>
-                        new HistoricDataModel({
-                            timestamp: moment
-                                .utc(row.day)
-                                .format('yyyy-MM-DD HH:mm:ss'),
-                            value: row.last ?? '0',
-                        }),
-                ) ?? [];
-            return results;
+
+            if (!time) {
+                return (
+                    query?.map((row) =>
+                        this.createDayHistoricDataModel(row, '0'),
+                    ) ?? []
+                );
+            }
+
+            const needsGapFilling = query.some((row) => !row.last);
```

Comment:
we can use the same approach from `getValues24h` to do the gap filling

Reply:
updated